File: Projects/MYONN/1_Gentle_Start_with_C++/dog_class.cpp
//
// Gentle start of C++
// Filename: dog_class.cpp
//
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

class Dog
{
private:
    char name[32];
    int temperature;

public:
    // Constructor Initialize the class
    Dog(char* petname, int temp)
    {
        strcpy(name,petname);
        temperature = temp;
    }

    // Member function status()
    void status()
    {
        printf("Name: %s\n", name);
		printf("Temp: %d\n", temperature);
    }

    void setTemperature(int temp)
    {
        temperature = temp;
    }

    // This Dog can bark()
    void bark()
    {
        printf("woof!\n");
    }
};

int main()
{
    Dog sizzles((char*)"sizzles", 37);
    Dog mutleys((char*)"mutleys", 36);

	sizzles.status();
	sizzles.bark();

	mutleys.status();
	mutleys.bark();

	sizzles.setTemperature(40);
	sizzles.status();
}

File: Projects/MYONN/1_Gentle_Start_with_Python/dog_class.py
#
# Gentle start of Python
# 	Class for Dog object
# Filename: dog_class.py
#

class Dog:
	# Initializer or constructor of the class
	def __init__(self, petname, temp):
		self.name = petname
		self.temperature = temp
		pass

	# Member function status()
	def status(self):
		print("Name: ", self.name)
		print("Temp: ", self.temperature)
		pass

	def setTemperature(self, temp):
		self.temperature = temp
		pass

	# This Dog can bark()
	def bark(self):
		print("woof!")
		pass

	pass

if __name__=="__main__" :
	sizzles=Dog("sizzles", 37)
	mutleys=Dog("mutleys", 36)

	sizzles.status()
	sizzles.bark()

	mutleys.status()
	mutleys.bark()

	sizzles.setTemperature(40)
	sizzles.status()
